### DISCUSSION TITLE
Prevent netlab from autostarting a lab on host boot

### DISCUSSION THREAD

**Discussion by matthoug** [ipspace/netlab#1321] [Q&A]
Hello,
I'm discovering this wonderful tool ! Thank you Ivan for the work,

I'm using netlab on a public cloud instance that I turn off whenever i don't need it, however when powering on the instance netlab will automatically start the latest lab, is here a way to prevent it ?

I've installed netlab using the regular netlab install procedure on a brand new 22.04 ubuntu image.

Regards,

**Comment by ipspace** [comment]
There's nothing in _netlab_ that would automatically start after the reboot. Also, are we talking about a system shutdown/poweroff (initiated from Linux CLI or similar), or a VM poweroff (that could leave the system in some interim state)?

Anyway, the artifacts from the last lab are still around, and Docker or libvirt might decide to restart the workload that was running at the time the system was powered off. It's a good practice to shut down all labs before shutting down the server.

To recover from the "some labs might still be running" state, use **netlab status --all** to display all running instances, and then **netlab status --cleanup --instance _id_** to shut down the lab and clean it up.

If you get to a state where the **netlab status --cleanup** fails to do its job (for example, the virtualization provider throws an error which stops the shutdown process), use **netlab status --all** to find the directory with a leftover lab, change into that directory, and do **netlab down --cleanup --force**.

Hope this helps (and yes, it should go into the documentation)

**Comment by jbemmel** [comment]
For Docker instances, I believe the auto-starting may be due to https://docs.docker.com/engine/containers/start-containers-automatically/

It doesn't happen for VM based devices for example, and there are flags in Containerlab that could be set to prevent this from happening

**Reply by ipspace** [reply]
I could set those flags, but it doesn't matter anyway. The container and VM artifacts are not cleaned up during the reboot (containers are still defined, whether they are running or stopped, and so are the VMs and their disks), so it's mandatory to do a cleanup anyway.

Will add a warning to **netlab up** documentation, or do you think there's a better place to do it? /cc @ssasso @steffann

**Comment by ipspace** [comment]
Fixed the containerlab behavior and a bunch of netlab issues. Lab restart after a reboot is now consistent (and documented)
